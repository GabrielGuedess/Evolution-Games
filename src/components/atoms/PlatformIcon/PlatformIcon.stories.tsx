import { Story, Meta } from '@storybook/react/types-6-0';

import { PlatformIcon, PlatformIconProps } from './PlatformIcon';

export default {
  title: 'Atoms/PlatformIcon',
  component: PlatformIcon,
  args: {
    hasTitle: true,
    isActive: false,
    size: 'medium',
    variant: 'simple',
  },
} as Meta<PlatformIconProps>;

export const Playstation4: Story<PlatformIconProps> = args => (
  <PlatformIcon {...args} />
);

Playstation4.args = {
  platform: 'ps4',
  color: 'white',
};

export const Playstation5: Story<PlatformIconProps> = args => (
  <PlatformIcon {...args} />
);

Playstation5.args = {
  platform: 'ps5',
  color: 'white',
};

export const XboxOne: Story<PlatformIconProps> = args => (
  <PlatformIcon {...args} />
);

XboxOne.args = {
  platform: 'one',
  color: 'white',
};

export const XboxSeries: Story<PlatformIconProps> = args => (
  <PlatformIcon {...args} />
);

XboxSeries.args = {
  platform: 'xs',
  color: 'white',
};

export const PC: Story<PlatformIconProps> = args => <PlatformIcon {...args} />;

PC.args = {
  platform: 'pc',
  color: 'white',
};

export const All: Story<PlatformIconProps> = args => <PlatformIcon {...args} />;

All.args = {
  platform: 'all',
  color: 'white',
};
