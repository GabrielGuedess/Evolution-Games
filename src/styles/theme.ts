export default {
  grid: {
    container: '165.6rem',
  },
  border: {
    radius: {
      regular: '1rem',
    },
  },
  font: {
    family: {
      primary:
        "'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    },
    regular: 400,
    medium: 500,
    sizes: {
      xxsmall: '1rem',
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '1.8rem',
      xxmedium: '2.0rem',
      big: '2.2rem',
      xbig: '2.5rem',
      xxbig: '2.8rem',
      xxxbig: '3.0rem',
      large: '3.2rem',
      xlarge: '4.5rem',
      xxlarge: '5.0rem',
      extra: '6.0rem',
    },
  },
  colors: {
    primary: '#7C3AED',
    secondary: '#717B8A',
    mainBg: '#04040C',
    white: '#FFFFFF',
    whiteText: '#F4F4F5',
    gameDetails: '#010914',
    gameGreat: '#0FBF00',
    gameAverage: '#FFF720',
    gameBad: '#FF3333',
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    big: '5.8rem',
    huge: '8.0rem',
    section: '10rem',
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
  shadows: {
    text: '0 0.8rem 1rem rgba(118, 74, 241, 0.5)',
    box: '0 0.8rem 2.8rem rgba(118, 74, 241, 0.5)',
  },
} as const;
