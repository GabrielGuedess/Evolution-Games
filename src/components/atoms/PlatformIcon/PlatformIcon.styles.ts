import styled, { css } from 'styled-components';

import { PlatformIconProps } from './PlatformIcon';

const iconModifiers = {
  small: () => ({
    playstation: () => css`
      width: 1.8rem;
      height: 1.8rem;
    `,

    xbox: () => css`
      width: 1.8rem;
      height: 1.8rem;
    `,

    pc: () => css`
      width: 2rem;
      height: 1rem;
    `,

    all: () => css`
      width: 1.5rem;
      height: 1.5rem;
    `,
  }),

  medium: () => ({
    playstation: () => css`
      width: 2.8rem;
      height: 2.8rem;
    `,

    xbox: () => css`
      width: 2.2rem;
      height: 2.2rem;
    `,

    pc: () => css`
      width: 3.2rem;
      height: 1.7rem;
    `,

    all: () => css`
      width: 2rem;
      height: 2rem;
    `,
  }),
};

export const Wrapper = styled.div<
  Pick<PlatformIconProps, 'color' | 'isActive'>
>`
  ${({ theme, color, isActive }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    color: ${isActive ? theme.colors.primary : theme.colors[color!]};
    transition: color 0.3s linear;

    :hover {
      cursor: pointer;
      color: ${theme.colors.primary};
    }
  `}
`;

export const Icon = styled.svg<Pick<PlatformIconProps, 'platform' | 'size'>>`
  ${({ platform, size }) => css`
    ${!!iconModifiers[size!]()[platform] && iconModifiers[size!]()[platform]}
  `}
`;

export const Label = styled.span<Pick<PlatformIconProps, 'size'>>`
  ${({ theme, size }) => css`
    font-style: normal;
    font-weight: ${theme.font.regular};
    margin-left: ${theme.spacings.xsmall};
    font-size: ${size === 'medium'
      ? theme.font.sizes.medium
      : theme.font.sizes.small};
  `}
`;
