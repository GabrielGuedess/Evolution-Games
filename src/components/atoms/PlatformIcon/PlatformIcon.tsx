import * as S from './PlatformIcon.styles';

export type Platform = 'playstation' | 'xbox' | 'pc' | 'all';

export type PlatformIconProps = {
  platform: Platform;
  color?: 'white' | 'secondary';
  hasTitle?: boolean;
  isActive?: boolean;
  onClick?: () => void;
};

export const PlatformIcon = ({
  onClick,
  platform,
  color = 'white',
  hasTitle = false,
  isActive = false,
}: PlatformIconProps) => {
  const platformTitle = {
    playstation: 'PlayStation',
    xbox: 'Xbox',
    pc: 'Personal computer',
    all: 'All platforms',
  };

  return (
    <S.Wrapper color={color} isActive={isActive} onClick={onClick}>
      {platform === 'playstation' && (
        <S.Icon
          fill="none"
          role="figure"
          platform={platform}
          viewBox="0 0 28 22"
          aria-label="PlayStation Icon"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill="currentColor"
            d="M.567 16.039c-1.05.71-.7 2.057 1.538 2.697 2.309.781 4.826.994 7.276.569.139 0 .28-.072.35-.072v-2.415l-2.378.781c-.91.284-1.818.356-2.727.141-.7-.213-.561-.638.28-.994l4.826-1.703v-2.624l-6.717 2.342c-.839.284-1.68.71-2.448 1.278ZM16.796 5.393v6.883c2.868 1.42 5.106 0 5.106-3.69 0-3.76-1.329-5.463-5.177-6.81C14.697 1.062 12.598.425 10.5 0v20.51L15.396 22V4.755c0-.782 0-1.348.561-1.135.77.212.839.994.839 1.773Zm9.096 9.012c-2.028-.71-4.197-.994-6.296-.781-1.119.071-2.17.355-3.149.71l-.21.07v2.77l4.549-1.704c.909-.284 1.818-.356 2.727-.14.7.212.56.637-.28.993l-6.996 2.626v2.697l9.655-3.62c.7-.284 1.328-.637 1.889-1.206.49-.712.28-1.706-1.89-2.415Z"
          />
        </S.Icon>
      )}

      {platform === 'xbox' && (
        <S.Icon
          fill="none"
          role="figure"
          platform={platform}
          viewBox="0 0 20 20"
          aria-label="Xbox Icon"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill="currentColor"
            d="M4.348 9.664c1.736-2.564 3.408-4.06 3.408-4.06s-1.842-2.14-4.06-2.948l-.144-.036A9.786 9.786 0 0 0 .2 10c0 2.218.74 4.266 1.978 5.906 0-.192.026-3.076 2.17-6.242ZM19.8 10c0-2.944-1.3-5.582-3.352-7.38l-.14.04c-2.218.81-4.064 2.948-4.064 2.948s1.672 1.496 3.408 4.06c2.144 3.168 2.17 6.05 2.166 6.244A9.747 9.747 0 0 0 19.8 10ZM7.602 2.34c1.282.574 2.388 1.514 2.388 1.514s1.116-.936 2.398-1.514c1.61-.722 2.84-.494 3.164-.416A9.77 9.77 0 0 0 10 .2a9.77 9.77 0 0 0-5.552 1.724c.316-.078 1.54-.31 3.154.416ZM14.494 11.614c-2.026-2.488-4.502-4.108-4.502-4.108s-2.464 1.618-4.494 4.108c-1.742 2.14-2.388 3.71-2.738 4.716l-.056.21A9.769 9.769 0 0 0 10 19.8c2.9 0 5.504-1.26 7.298-3.26l-.062-.21c-.35-1.006-1.002-2.576-2.742-4.716Z"
          />
        </S.Icon>
      )}

      {platform === 'pc' && (
        <S.Icon
          fill="none"
          role="figure"
          platform={platform}
          viewBox="0 0 34 19"
          aria-label="PC Icon"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill="currentColor"
            d="M.426 18 3.324.545h6.886c1.318 0 2.4.253 3.247.759.853.5 1.45 1.196 1.79 2.088.347.886.426 1.91.239 3.068-.193 1.165-.614 2.19-1.262 3.077a6.556 6.556 0 0 1-2.505 2.062c-1.029.495-2.216.742-3.563.742H3.767l.486-2.957h3.792c.716 0 1.327-.123 1.833-.367.511-.25.918-.594 1.219-1.031.3-.443.497-.952.588-1.526.096-.585.065-1.094-.094-1.525a1.87 1.87 0 0 0-.878-1.006c-.42-.244-.986-.366-1.696-.366H6.528L4.116 18H.426ZM33.232 6.656H29.55c.012-.488-.05-.92-.187-1.295a2.533 2.533 0 0 0-.614-.972 2.576 2.576 0 0 0-.98-.605 3.76 3.76 0 0 0-1.304-.213c-.932 0-1.787.233-2.565.699-.779.466-1.435 1.142-1.969 2.028-.534.881-.903 1.946-1.108 3.196-.199 1.205-.187 2.213.034 3.026.222.812.614 1.426 1.176 1.84.569.41 1.276.614 2.122.614a5.56 5.56 0 0 0 1.492-.196c.472-.136.9-.33 1.287-.58a4.309 4.309 0 0 0 1.023-.928c.295-.364.525-.776.69-1.236h3.707a8.442 8.442 0 0 1-1.09 2.318 8.73 8.73 0 0 1-1.816 1.995 8.76 8.76 0 0 1-2.463 1.38c-.92.341-1.929.512-3.026.512-1.585 0-2.945-.364-4.082-1.091-1.13-.727-1.946-1.776-2.446-3.145-.5-1.37-.594-3.02-.281-4.952.312-1.87.923-3.454 1.832-4.756.915-1.306 2.026-2.298 3.333-2.974a9.063 9.063 0 0 1 4.21-1.014c1.04 0 1.98.142 2.82.426.842.284 1.558.699 2.149 1.244a5.098 5.098 0 0 1 1.346 1.986c.301.784.432 1.682.392 2.693Z"
          />
        </S.Icon>
      )}

      {platform === 'all' && (
        <S.Icon
          fill="none"
          role="figure"
          platform={platform}
          viewBox="0 0 20 20"
          aria-label="All Platforms Icon"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g fill="currentColor" clipPath="url(#a)">
            <path d="M6.902 0H2.297C1.028 0 0 1.017 0 2.271v4.554C0 8.08 1.028 9.097 2.297 9.097h4.605c1.269 0 2.297-1.017 2.297-2.272V2.271C9.199 1.017 8.17 0 6.902 0ZM17.709 0h-4.605c-1.269 0-2.297 1.017-2.297 2.271v4.554c0 1.255 1.028 2.272 2.297 2.272h4.605c1.268 0 2.297-1.017 2.297-2.272V2.271C20.006 1.017 18.977 0 17.709 0ZM6.902 10.903H2.297C1.028 10.903 0 11.92 0 13.174v4.555C0 18.983 1.028 20 2.297 20h4.605c1.269 0 2.297-1.017 2.297-2.271v-4.554c0-1.255-1.028-2.272-2.297-2.272ZM17.709 10.903h-4.605c-1.269 0-2.297 1.017-2.297 2.271v4.555c0 1.254 1.028 2.271 2.297 2.271h4.605c1.268 0 2.297-1.017 2.297-2.271v-4.554c0-1.255-1.029-2.272-2.297-2.272Z" />
          </g>
        </S.Icon>
      )}

      {hasTitle && !!platformTitle[platform] && (
        <S.Label aria-label={platformTitle[platform]}>
          {platformTitle[platform]}
        </S.Label>
      )}
    </S.Wrapper>
  );
};
