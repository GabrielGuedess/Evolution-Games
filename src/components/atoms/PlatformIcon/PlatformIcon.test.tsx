import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import theme from 'styles/theme';

import { PlatformIcon } from './PlatformIcon';

describe('<PlatformIcon />', () => {
  it('should render the playstation icon', () => {
    const { container } = renderWithTheme(
      <PlatformIcon platform="playstation" />,
    );

    // Arrange
    const icon = screen.getByLabelText('PlayStation Icon');

    // Assert
    expect(icon).toBeInTheDocument();
    expect(icon.parentElement).toHaveStyle({ color: theme.colors.white });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render selected playstation icon', () => {
    const { container } = renderWithTheme(
      <PlatformIcon platform="playstation" hasTitle isActive />,
    );

    // Arrange
    const icon = screen.getByLabelText('PlayStation Icon');
    const label = screen.getByLabelText('PlayStation');

    // Assert
    expect(label.parentElement).toHaveStyle({
      color: theme.colors.primary,
    });

    expect(icon.parentElement).toHaveStyle({
      color: theme.colors.primary,
    });

    expect(icon).toHaveStyle({
      height: '2.8rem',
      width: '2.8rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render pc icon in secondary color', () => {
    const { container } = renderWithTheme(
      <PlatformIcon platform="pc" color="secondary" hasTitle />,
    );

    // Arrange
    const icon = screen.getByLabelText('PC Icon');
    const label = screen.getByLabelText('Personal computer');

    // Assert
    expect(label.parentElement).toHaveStyle({
      color: theme.colors.secondary,
    });

    expect(icon.parentElement).toHaveStyle({
      color: theme.colors.secondary,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
