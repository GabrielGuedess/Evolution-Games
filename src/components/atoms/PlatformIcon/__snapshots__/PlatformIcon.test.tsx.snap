// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<PlatformIcon /> should render pc icon in secondary color 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #717B8A;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c0:hover {
  cursor: pointer;
  color: #855aff;
}

.c1 {
  width: 3.2rem;
  height: 1.7rem;
}

.c2 {
  font-size: 1.6rem;
  font-style: normal;
  font-weight: 400;
  margin-left: 1.6rem;
}

<div
  class="c0"
  color="secondary"
>
  <svg
    aria-label="PC Icon"
    class="c1"
    fill="none"
    role="figure"
    viewBox="0 0 34 19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M.426 18 3.324.545h6.886c1.318 0 2.4.253 3.247.759.853.5 1.45 1.196 1.79 2.088.347.886.426 1.91.239 3.068-.193 1.165-.614 2.19-1.262 3.077a6.556 6.556 0 0 1-2.505 2.062c-1.029.495-2.216.742-3.563.742H3.767l.486-2.957h3.792c.716 0 1.327-.123 1.833-.367.511-.25.918-.594 1.219-1.031.3-.443.497-.952.588-1.526.096-.585.065-1.094-.094-1.525a1.87 1.87 0 0 0-.878-1.006c-.42-.244-.986-.366-1.696-.366H6.528L4.116 18H.426ZM33.232 6.656H29.55c.012-.488-.05-.92-.187-1.295a2.533 2.533 0 0 0-.614-.972 2.576 2.576 0 0 0-.98-.605 3.76 3.76 0 0 0-1.304-.213c-.932 0-1.787.233-2.565.699-.779.466-1.435 1.142-1.969 2.028-.534.881-.903 1.946-1.108 3.196-.199 1.205-.187 2.213.034 3.026.222.812.614 1.426 1.176 1.84.569.41 1.276.614 2.122.614a5.56 5.56 0 0 0 1.492-.196c.472-.136.9-.33 1.287-.58a4.309 4.309 0 0 0 1.023-.928c.295-.364.525-.776.69-1.236h3.707a8.442 8.442 0 0 1-1.09 2.318 8.73 8.73 0 0 1-1.816 1.995 8.76 8.76 0 0 1-2.463 1.38c-.92.341-1.929.512-3.026.512-1.585 0-2.945-.364-4.082-1.091-1.13-.727-1.946-1.776-2.446-3.145-.5-1.37-.594-3.02-.281-4.952.312-1.87.923-3.454 1.832-4.756.915-1.306 2.026-2.298 3.333-2.974a9.063 9.063 0 0 1 4.21-1.014c1.04 0 1.98.142 2.82.426.842.284 1.558.699 2.149 1.244a5.098 5.098 0 0 1 1.346 1.986c.301.784.432 1.682.392 2.693Z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-label="Personal computer"
    class="c2"
  >
    Personal computer
  </span>
</div>
`;

exports['<PlatformIcon /> should render selected playstation icon 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #855aff;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c0:hover {
  cursor: pointer;
  color: #855aff;
}

.c1 {
  width: 2.8rem;
  height: 2.8rem;
}

.c2 {
  font-size: 1.6rem;
  font-style: normal;
  font-weight: 400;
  margin-left: 1.6rem;
}

<div
  class="c0"
  color="white"
>
  <svg
    aria-label="Playstation Icon"
    class="c1"
    fill="none"
    role="figure"
    viewBox="0 0 28 22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M.567 16.039c-1.05.71-.7 2.057 1.538 2.697 2.309.781 4.826.994 7.276.569.139 0 .28-.072.35-.072v-2.415l-2.378.781c-.91.284-1.818.356-2.727.141-.7-.213-.561-.638.28-.994l4.826-1.703v-2.624l-6.717 2.342c-.839.284-1.68.71-2.448 1.278ZM16.796 5.393v6.883c2.868 1.42 5.106 0 5.106-3.69 0-3.76-1.329-5.463-5.177-6.81C14.697 1.062 12.598.425 10.5 0v20.51L15.396 22V4.755c0-.782 0-1.348.561-1.135.77.212.839.994.839 1.773Zm9.096 9.012c-2.028-.71-4.197-.994-6.296-.781-1.119.071-2.17.355-3.149.71l-.21.07v2.77l4.549-1.704c.909-.284 1.818-.356 2.727-.14.7.212.56.637-.28.993l-6.996 2.626v2.697l9.655-3.62c.7-.284 1.328-.637 1.889-1.206.49-.712.28-1.706-1.89-2.415Z"
      fill="currentColor"
    />
  </svg>
  <span
    aria-label="Playstation"
    class="c2"
  >
    Playstation
  </span>
</div>
`;

exports['<PlatformIcon /> should render the playstation icon 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FFFFFF;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c0:hover {
  cursor: pointer;
  color: #855aff;
}

.c1 {
  width: 2.8rem;
  height: 2.8rem;
}

<div
  class="c0"
  color="white"
>
  <svg
    aria-label="Playstation Icon"
    class="c1"
    fill="none"
    role="figure"
    viewBox="0 0 28 22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M.567 16.039c-1.05.71-.7 2.057 1.538 2.697 2.309.781 4.826.994 7.276.569.139 0 .28-.072.35-.072v-2.415l-2.378.781c-.91.284-1.818.356-2.727.141-.7-.213-.561-.638.28-.994l4.826-1.703v-2.624l-6.717 2.342c-.839.284-1.68.71-2.448 1.278ZM16.796 5.393v6.883c2.868 1.42 5.106 0 5.106-3.69 0-3.76-1.329-5.463-5.177-6.81C14.697 1.062 12.598.425 10.5 0v20.51L15.396 22V4.755c0-.782 0-1.348.561-1.135.77.212.839.994.839 1.773Zm9.096 9.012c-2.028-.71-4.197-.994-6.296-.781-1.119.071-2.17.355-3.149.71l-.21.07v2.77l4.549-1.704c.909-.284 1.818-.356 2.727-.14.7.212.56.637-.28.993l-6.996 2.626v2.697l9.655-3.62c.7-.284 1.328-.637 1.889-1.206.49-.712.28-1.706-1.89-2.415Z"
      fill="currentColor"
    />
  </svg>
</div>
`;
